File: tut/src/service/adder_srv.py
#!/usr/bin/env python3

import rospy as ros
from tut.srv import *


def adderCallback(request):
    response = AdderResponse()
    response.sum.num = request.a.num + request.b.num
    return response;


def main():
    ros.init_node("adder_srvpy")
    ros.Service("/adder", Adder, adderCallback)
    ros.loginfo("Adder ready!")
    ros.spin()


if (__name__ == "__main__"):
    main()

File: tut/src/service/adder_srv.cpp
#include <ros/ros.h>
#include "tut/Adder.h"


bool add(tut::AdderRequest &request, tut::AdderResponse &responce)
{
    responce.sum.num = request.a.num + request.b.num;
    return true;
}


int main(int argc, char **argv)
{
    ros::init(argc, argv, "adder_srv");
    ros::NodeHandle nh;
    ros::ServiceServer server = nh.advertiseService("/adder", add);
    ros::Duration(1).sleep();

    ROS_INFO("Adder server ready!");
    
    ros::spin();

    return 0;
}
